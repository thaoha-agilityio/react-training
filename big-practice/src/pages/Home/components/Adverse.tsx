import { Box, Flex, Heading, Image } from '@chakra-ui/react';

// Images
import adverse from '@assets/photos/adverse.png';

export const Adverse = (): JSX.Element => (
  <Box my='50px' as='section'>
    <Flex justifyContent='center'>
      <Heading
        as='h3'
        fontWeight='bolder'
        fontSize={{ base: 'lg', md: '3xl' }}
        lineHeight='48px'
        textShadow='0px 4px 4px rgba(0, 0, 0, 0.25)'
        wordBreak='break-word'
        textAlign='center'
        w='240px'
      >
        #FuniroFurniture
      </Heading>
    </Flex>
    <Image src={adverse} alt='adverse' width='100%' height='100%' />
  </Box>
);
