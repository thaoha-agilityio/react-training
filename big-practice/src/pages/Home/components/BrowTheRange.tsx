import { memo } from 'react';
import { Box, Heading, Text, Image, Flex, Stack } from '@chakra-ui/react';

//Images
import dining from '@assets/photos/dining.jpg';
import bedroom from '@assets/photos/bedroom.jpg';
import living from '@assets/photos/living.jpg';

export const BrowTheRange = memo(
  (): JSX.Element => (
    <Box as='section'>
      <Box textAlign='center' textShadow={'0px 4px 4px rgba(0, 0, 0, 0.25)'} mt='40px'>
        <Heading
          as='h3'
          fontSize={{ base: 'lg', md: 'xl' }}
          fontWeight='bold'
          color='secondary.500'
        >
          Browse The Range
        </Heading>
        <Text fontSize={{ base: 'sm', md: 'md' }} color='secondary.200'>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </Text>
      </Box>
      <Flex
        fontSize='lg'
        color='secondary.500'
        mt='40px'
        gap='20px'
        justify='center'
        wrap={{ base: 'wrap', md: 'nowrap' }}
        textShadow={'0px 4px 4px rgba(0, 0, 0, 0.25)'}
      >
        <Stack w='480px' textAlign='center'>
          <Image src={dining} borderRadius='xl' alt='dining' w='100%' h='100%' />
          <Text>Dining</Text>
        </Stack>
        <Stack w='480px' textAlign='center'>
          <Image src={bedroom} borderRadius='xl' alt='bedroom' w='100%' h='100%' />
          <Text>Bedroom</Text>
        </Stack>
        <Stack w='480px' textAlign='center'>
          <Image src={living} borderRadius='xl' alt='living' w='100%' h='100%' />
          <Text>Living</Text>
        </Stack>
      </Flex>
    </Box>
  ),
);
