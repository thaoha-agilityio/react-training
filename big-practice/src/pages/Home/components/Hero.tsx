import { memo } from 'react';
import { Box, Button, Heading, Text } from '@chakra-ui/react';

// Images
import hero from '@assets/photos/hero.png';
import heroWepb from '@assets/photos/hero.webp';

export const Hero = memo(
  (): JSX.Element => (
    <Box as='section' pos='relative' h={{ base: '500px', md: '1070px' }}>
      <Box
        bgImage={[`url(${hero})`, `url(${heroWepb})`]}
        bgSize='cover'
        w='100%'
        height='100%'
        bgPos='center'
      ></Box>

      <Box
        w={{ base: '300px', md: '643px' }}
        h={{ base: '250px', md: '443px' }}
        borderRadius='md'
        bg='yellow.200'
        border='1px solid'
        pos='absolute'
        transform={{ base: 'translateX(-50%)', lg: 'translateX(0%)' }}
        left='50%'
        top='37%'
      >
        <Box pl='40px' pt={{ base: '25px', md: '62px' }}>
          <Text
            textTransform='uppercase'
            letterSpacing='3px'
            fontSize='sm'
            color='gray.500'
            fontWeight='bold'
          >
            new arrival
          </Text>
          <Heading
            textTransform='capitalize'
            color='yellow.250'
            textShadow={'0px 4px 4px rgba(0, 0, 0, 0.25)'}
            fontWeight='bolder'
            fontSize={{ base: 'xxl', md: '5xl' }}
          >
            discover our new collection
          </Heading>
          <Text
            variant='primary'
            fontWeight='bold'
            lineHeight='sm'
            pt='17px'
            py={{ base: '10px', md: '36px' }}
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut elit tellus, luctus nec
            ullamcorper mattis.
          </Text>
          <Button
            variant='solid'
            textTransform='uppercase'
            fontWeight='bolder'
            w={{ base: '100px', md: '222px' }}
            h={{ base: '30px', md: '74px' }}
            fontSize={{ base: 'tiny', md: 'base' }}
          >
            buy now
          </Button>
        </Box>
      </Box>
    </Box>
  ),
);
