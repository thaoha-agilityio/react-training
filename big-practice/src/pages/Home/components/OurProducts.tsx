import { Flex, Heading, Spinner, Stack, Text } from '@chakra-ui/react';

// Components
import Products from '@components/Products';

// Custom hooks
import { useFetchProducts } from '@hooks/useProduct';

//  Stores
import { useProductStore } from '@stores';

export const OurProducts = () => {
  const products = useProductStore((state) => state.products);
  const { isLoading, isError, error } = useFetchProducts();

  return (
    <Stack as='section' my='50px' justifyContent='center'>
      <Heading
        as='h3'
        fontSize={{ base: 'lg', md: '3xl' }}
        color='gray.250'
        fontWeight='bolder'
        textShadow={'0px 4px 4px rgba(0, 0, 0, 0.25)'}
        textAlign='center'
      >
        Our Products
      </Heading>
      <Flex pt='32px' gap='32px' wrap='wrap' justifyContent='center'>
        {/* TODO: update latter */}
        {isError ? <Text>{error.message}</Text> : <Products products={products} />}
        {isLoading && <Spinner />}
      </Flex>
    </Stack>
  );
};
