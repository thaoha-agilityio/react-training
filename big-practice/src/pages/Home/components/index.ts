export * from './Hero';
export * from './BrowTheRange';
export * from './Adverse';
export * from './OurProducts';
