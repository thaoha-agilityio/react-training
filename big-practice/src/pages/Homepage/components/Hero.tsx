import { Box, Button, Heading, Image, Text } from '@chakra-ui/react';

import hero from '@assets/photos/hero.png';

export const Hero = () => (
  <section>
    <Box pos='relative'>
      <Image src={hero} w='100%' />
    </Box>
    <Box w='643px' h='443px' borderRadius='md' bg='yellow.200' border='1px solid' pos='absolute'>
      <Box pl='40px' pt='62px'>
        <Text
          textTransform='uppercase'
          letterSpacing='3px'
          fontSize='sm'
          color='gray.500'
          fontWeight='bold'
        >
          new arrival
        </Text>
        <Heading
          as='h1'
          textTransform='capitalize'
          color='yellow.250'
          textShadow={'0px 4px 4px rgba(0, 0, 0, 0.25)'}
          fontWeight='bolder'
          fontSize={{ base: '', md: '4xl' }}
        >
          discover our new collection
        </Heading>
        <Text fontWeight='bold' fontSize='base' lineHeight='sm' pt='17px' py='36px'>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut elit tellus, luctus nec
          ullamcorper mattis.
        </Text>
        <Button textTransform='uppercase' fontWeight='bolder' variant='solid' w='222px' h='74px'>
          buy now
        </Button>
      </Box>
    </Box>
  </section>
);
