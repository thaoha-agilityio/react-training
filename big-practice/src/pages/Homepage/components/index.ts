export * from './Hero';
export * from './BrowserTheRange';
export * from './FuniroFurniture';
export * from './OurProducts';
