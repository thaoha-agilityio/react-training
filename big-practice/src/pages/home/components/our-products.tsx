import { Button, Flex, Heading, Stack } from '@chakra-ui/react';
import { Pagination } from '@components/Pagination';
import { Products } from '@components/Products';

import { useProductStore } from '@stores';

export const OurProducts = () => {
  const products = useProductStore((state) => state.products);

  return (
    <section>
      <Stack my='50px' justifyContent='center'>
        <Heading
          as='h3'
          fontSize={{ base: 'lg', md: '3xl' }}
          color='gray.250'
          fontWeight='bolder'
          textShadow={'0px 4px 4px rgba(0, 0, 0, 0.25)'}
          textAlign='center'
        >
          Our Products
        </Heading>
        <Flex pt='32px' gap='32px' wrap='wrap' justifyContent='center'>
          {/* TODO: update latter */}
          <Products products={products} />
        </Flex>
        <Flex pt='32px' justifyContent='center'>
          <Button w='245px' h='48px'>
            Show More
          </Button>
        </Flex>
      </Stack>
      <Pagination />
    </section>
  );
};
