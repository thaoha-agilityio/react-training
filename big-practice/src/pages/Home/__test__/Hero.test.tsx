import { render } from '@testing-library/react';
import { Hero } from '../components';

describe('Hero component', () => {
  it('should render component correctly', () => {
    const { container } = render(<Hero />);

    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
