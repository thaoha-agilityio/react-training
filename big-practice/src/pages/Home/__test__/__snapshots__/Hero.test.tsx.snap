// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component should render component correctly 1`] = `
<div>
  <section
    class="css-1x1cs40"
  >
    <div
      class="css-1rrw7j4"
    />
    <div
      class="css-1nio0lb"
    >
      <div
        class="css-1vvasvl"
      >
        <p
          class="chakra-text css-b2hpjo"
        >
          New Arrival
        </p>
        <h2
          class="chakra-heading css-sbfz88"
        >
          discover our new collection
        </h2>
        <p
          class="chakra-text css-152l52r"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut elit tellus, luctus nec ullamcorper mattis.
        </p>
        <button
          class="chakra-button css-1gvc6wf"
          type="button"
        >
          buy now
        </button>
      </div>
    </div>
  </section>
</div>
`;
